test/e2e/chainsaw/check-multi-resource-rgd/instance-create-assert.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   generation: 1
   labels:
     kro.run/owned: "true"
-    kro.run/resource-graph-definition-name: check-multi-resource-rgd
+    kro.run/source-resource-graph-definition-name: check-multi-resource-rgd
```

Comment:
this looks a bit off, is this a breaking change? maybe lacking about the understanding of these labels is my issue here, feel free to resolve if non-issue

Reply:
When testing i found the case where we may have recursive RGDs. RGD1 having RGD2 instance as part of its resources. In such a case, RGD2's instance created by RGD1 would have 2 sets of labels. 

The source one which points to the RGD2 as its source definition.
The regular one which points to RGD1 instance as its creator.